projects/laji/src/app/+observation/observation.facade.ts

```diff
@@ -126,17 +121,26 @@ export class ObservationFacade {
             query[key] =  loggedIn ? this.userService.getToken() : undefined;
           }
         });
+
+        const newQuery = Util.clone(query);
+
         const hash = JSON.stringify(warehouseQuery);
-        if (this.hashCache['query'] === hash) {
+        if (this.queryHash === hash) {
+          this.updateState({..._state, newQuery, newQueryHasChanges: false});
           return;
         }
-        this.hashCache['query'] = hash;
+        this.queryHash = hash;
 
-        this.updateState({..._state, query, loadingUnits: true, loadingTaxa: true});
+        this.updateState({..._state, query, newQuery, newQueryHasChanges: false, loadingUnits: true, loadingTaxa: true});
```

Comment:
The state change between the if-condition and the "else part" would be easier to understand if a single state object would be binded to a variable before the if-block, and the "else" part would add stuff to the state var.
```
const nextState ={..._state, newQuery, newQueryHasChanges: false};
       if (this.queryHash === hash) {
          this.updateState(nextState);
          return;
        }
        this.queryHash = hash;

        this.updateState({...nextState, loadingUnits: true, loadingTaxa: true});
```